### PR TITLE
feat: [CDS-34607]: pipeline deployable per env GQL changes

### PR DESCRIPTION
**Pull request by abhinav-harness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
…s-core into abhinav/CDS-34607

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31759)

```
mutation($userGroup: UpdateUserGroupPermissionsInput!){
  updateUserGroupPermissions (input:$userGroup) {
      permissions {
        accountPermissions{
          accountPermissionTypes
        }
        appPermissions {
          permissionType
         
          deployments {
            filterTypes
            envIds
            executableElementFilter{
              executableElementType
              executableElementFilterType
              ids
            }
          }
       
          actions
        }
      }
    }
  }
```

```
{
  "userGroup": {
    "userGroupId": "tv2JQ-wuSi61w7J7PotQ9A",
    "permissions": {
      "accountPermissions": {
       "accountPermissionTypes":  ["READ_USERS_AND_GROUPS","MANAGE_USERS_AND_GROUPS","MANAGE_TAGS","ADMINISTER_OTHER_ACCOUNT_FUNCTIONS","MANAGE_TEMPLATE_LIBRARY","VIEW_AUDITS"]
      },
      "appPermissions": [
        {
                "permissionType": "DEPLOYMENT",
                "actions": [
                  "EXECUTE_PIPELINE",
                  "READ",
                  "EXECUTE_WORKFLOW",
                  "ROLLBACK_WORKFLOW"
                ],
                "applications": {
                  "appIds": [
                    "R_zddPb3RLucSftpgtLp5A"
                  ]
                },
                "deployments": {
                  "envIds": null,
                  "executableElementFilter": {
                    "executableElementType": "PIPELINE",
                    "executableElementFilterType": "ALL",
                    "ids": [
                      "pSClsQs_QPOOLOavJXaHsA"
                    ]
                  },
                  "filterTypes": [
                    "NON_PRODUCTION_ENVIRONMENTS"
                  ]
                }
        }
      ]
    }
  }
}
```

Screenshot 2022-03-26 at 3 46 56 AM

```
query{
  userGroups(limit:10){
    nodes{
      id
      permissions{
        appPermissions{
          permissionType
          actions
          applications{
            appIds
          }
          
          deployments{
        		
            envIds
            executableElementFilter{
              executableElementType
              executableElementFilterType
              ids
            }
            filterTypes
          }
        }
      }
    }
  }
}
```

```
query{
  userGroups(limit:10){
    nodes{
      id
      permissions{
        appPermissions{
          permissionType
          actions
          applications{
            appIds
          }
          
          deployments{
        		
            envIds
            executableElementFilter{
              executableElementType
              executableElementFilterType
              ids
            }
            filterTypes
          }
        }
      }
    }
  }
}
```

```
{
  "data": {
    "userGroups": {
      "nodes": [
        {
          "id": "tv2JQ-wuSi61w7J7PotQ9A",
          "permissions": {
            "appPermissions": [
              {
                "permissionType": "DEPLOYMENT",
                "actions": [
                  "EXECUTE_PIPELINE",
                  "READ",
                  "EXECUTE_WORKFLOW",
                  "ROLLBACK_WORKFLOW"
                ],
                "applications": {
                  "appIds": [
                    "R_zddPb3RLucSftpgtLp5A"
                  ]
                },
                "deployments": {
                  "envIds": null,
                  "executableElementFilter": {
                    "executableElementType": "PIPELINE",
                    "executableElementFilterType": null,
                    "ids": [
                      "pSClsQs_QPOOLOavJXaHsA"
                    ]
                  },
                  "filterTypes": [
                    "NON_PRODUCTION_ENVIRONMENTS"
                  ]
                }
              }
            ]
          }
        }
      ]
    }
  }
}
```

Screenshot 2022-03-26 at 3 47 24 AM

### CODE REVIEW

**Comment by hingerabhinav** [comment]
___
*[400-rest/src/main/java/software/wings/graphql/datafetcher/userGroup/UserGroupPermissionsController.java, line 364 at r1](https://reviewable.io/reviews/harness/harness-core/31759#-Mz-tmD68fitVKYgqy_V:-Mz-tmD68fitVKYgqy_W:b-u79yxx) ([raw file](https://github.com/harness/harness-core/blob/938e45a39a24132bc8d8e38ddbe6df32618b7de5/400-rest/src/main/java/software/wings/graphql/datafetcher/userGroup/UserGroupPermissionsController.java#L364)):*
> ```Java
>     }
>     if (deploymentPermissions.getExecutableElementFilter() == null) {
>       return EnvFilter.builder().ids(deploymentPermissions.getEnvIds()).filterTypes(filterTypes).build();
> ```

this should be WorkflowFilter?

File: 400-rest/src/main/resources/graphql/public/governance.graphql [hunk 1]
```diff
@@ -155,6 +155,28 @@ enum ServiceFilterType {
   CUSTOM
 }
 
+enum ExecutableElementType {
```
Comment (hingerabhinav) [inline 400-rest/src/main/resources/graphql/public/governance.graphql:158]: this should not be in governance. That is for deployment freeze filters

File: 400-rest/src/main/resources/graphql/public/governance.graphql [hunk 2]
```diff
@@ -155,6 +155,28 @@ enum ServiceFilterType {
...
+  WORKFLOW
+}
+
+enum ExecutableElementFilterType {
+  ALL
+}
+
+input ExecutableElementFilterInput {
+  executableElementType: ExecutableElementType!
+  ids: [String]
```
Comment (hingerabhinav) [inline 400-rest/src/main/resources/graphql/public/governance.graphql:170]: nit: [String!] would prevent giving empty array

Comment (abhinav-harness) [reply 400-rest/src/main/resources/graphql/public/governance.graphql]: He can give "ALL"

File: 400-rest/src/main/resources/graphql/public/governance.graphql (hunk 1 above)
Comment (abhinav-harness) [reply 400-rest/src/main/resources/graphql/public/governance.graphql]: done

**Comment by abhinav-harness** [comment]
___
*[400-rest/src/main/java/software/wings/graphql/datafetcher/userGroup/UserGroupPermissionsController.java, line 364 at r1](https://reviewable.io/reviews/harness/harness-core/31759#-Mz-tmD68fitVKYgqy_V:-Mz1UgsKD6Q4km4pZlhM:b32wsm) ([raw file](https://github.com/harness/harness-core/blob/938e45a39a24132bc8d8e38ddbe6df32618b7de5/400-rest/src/main/java/software/wings/graphql/datafetcher/userGroup/UserGroupPermissionsController.java#L364)):*
<details><summary><i>Previously, hingerabhinav (Abhinav Hinger) wrote…</i></summary><blockquote>

this should be WorkflowFilter?
</blockquote></details>

no

**Pull request merged**